internal/factsengine/gatherers/saptune.go

```diff
@@ -0,0 +1,128 @@
+package gatherers
+
+import (
+	"encoding/json"
+	"strings"
+
+	log "github.com/sirupsen/logrus"
+	"github.com/trento-project/agent/internal/core/saptune"
+	"github.com/trento-project/agent/pkg/factsengine/entities"
+	"github.com/trento-project/agent/pkg/utils"
+)
+
+const (
+	SaptuneGathererName = "saptune"
+)
+
+// nolint:gochecknoglobals
+var whitelistedArguments = map[string]string{
+	"status":          "status --non-compliance-check",
+	"solution-verify": "solution verify",
+	"solution-list":   "solution list",
+	"note-verify":     "note verify",
+	"note-list":       "note list",
+}
+
+// nolint:gochecknoglobals
+var (
+	SaptuneVersionUnsupported = entities.FactGatheringError{
+		Type:    "saptune-version-not-supported",
+		Message: "currently installed version of saptune is not supported",
+	}
+
+	SaptuneUnknownArgument = entities.FactGatheringError{
+		Type:    "saptune-unknown-error",
+		Message: "the requested argument is not currently supported",
+	}
+
+	SaptuneMissingArgument = entities.FactGatheringError{
+		Type:    "saptune-missing-argument",
+		Message: "missing required argument",
+	}
+
+	SaptuneCommandError = entities.FactGatheringError{
+		Type:    "saptune-cmd-error",
+		Message: "error executing saptune command",
+	}
+
+)
+
+type SaptuneGatherer struct {
+	executor utils.CommandExecutor
+}
+
+func NewDefaultSaptuneGatherer() *SaptuneGatherer {
+	return NewSaptuneGatherer(utils.Executor{})
+}
+
+func NewSaptuneGatherer(executor utils.CommandExecutor) *SaptuneGatherer {
+	return &SaptuneGatherer{
+		executor: executor,
+	}
+}
+
+func parseJSONToFactValue(jsonStr string) (entities.FactValue, error) {
+	// Unmarshal the JSON into an interface{} type.
+	var jsonData interface{}
+	if err := json.Unmarshal([]byte(jsonStr), &jsonData); err != nil {
+		return nil, err
+	}
+
+	// Convert the parsed jsonData into a FactValue using NewFactValue.
+	return entities.NewFactValueMod(jsonData)
```

Comment:
I don't think we need any new function like this.
We already have other gatherers converting a map to `FactValueMap`.
Check the `xml.go` file for example

Reply:
This is to avoid the default conversion that always converts "3" to a FactValueInt{Value: 3}. If we are ok with any quoted number to be converted to a FactValueInt, then we can remove it.

Reply:
We should be fine in my opinion, but if you want you could ask the checks team, as they will be the final users afterall